geomstats/geometry/discrete_curves.py

```diff
@@ -1842,7 +1844,326 @@ def horizontal_path(t):
 
         return horizontal_path
 
-    def align(self, point, base_point, n_times=20, threshold=1e-3):
+    def _dynamic_programming_single(
+            self, initial_curve, end_curve, n_discretization=100, max_slope=6):
+        r"""Compute the dynamic programming algorithm.
+
+        Find the reparametrization gamma of end_curve that minimizes the distance
+        between initial_curve and end_curve reparametrized by gamma, and output
+        the corresponding distance, using a dynamic programming algorithm.
+
+        The objective can be expressed in terms of square root velocity (SRV)
+        representations: it is equivalent to finding the gamma that maximizes
+        the L2 scalar product between initial_srv and end_srv@gamma where initial_srv
+        is the SRV representation of the initial curve and end_srv@gamma is the SRV
+        representation of the end curve reparametrized by gamma, i.e
+        .. math::
+        end_srv@\gamma(t) = end_srv(\gamma(t))\cdot|\gamma(t)|^\frac{1}{2}
+
+        The dynamic programming algorithm assumes that for every subinterval
+        :math: '\left[\frac{i}{n},\frac{i+1}{n}\right]' of :math: '\left[0,1\right]',
+        gamma is linear.
+
+        Inputs
+        ----------
+        intial_curve : array-like, shape=[k_sampling_points, ambient_dim]
+            Initial discrete curve.
+        end_curve : array-like, shape=[k_sampling_points, ambient_dim]
+            End discrete curve.
+        n_discretization : int
+            Number of subintervals in which the reparametrization is linear.
+            Optinal, default: 100.
+        max_slope : int
+            Maximum slope allowed for a reparametrization.
+            Optional, default: 6.
+
+
+        Outputs
+        -------
+        geodesic: callable
+            Time parametrized geodesic.
+        dist : float
+            Quotient distance between intial_curve and end_curve.
+
+        References
+        ----------
+        Mio, Washington & Srivastava, Anuj & Joshi, Shantanu. (2007).
```

Comment:
Thanks for adding the reference. Could you just copy the style used in geomstats ? See [here](https://github.com/geomstats/geomstats/blob/master/geomstats/geometry/discrete_curves.py#LL816C5-L816C15).